lib/ocpp/v201/smart_charging.cpp

```diff
@@ -598,16 +598,31 @@ CompositeSchedule SmartChargingHandler::calculate_composite_schedule(
         }
     }
 
-    auto charging_station_external_constraints = ocpp::v201::calculate_composite_schedule(
-        charging_station_external_constraints_periods, start_time, end_time, charging_rate_unit);
-    auto composite_charge_point_max =
-        ocpp::v201::calculate_composite_schedule(charge_point_max_periods, start_time, end_time, charging_rate_unit);
-    auto composite_tx_default =
-        ocpp::v201::calculate_composite_schedule(tx_default_periods, start_time, end_time, charging_rate_unit);
-    auto composite_tx = ocpp::v201::calculate_composite_schedule(tx_periods, start_time, end_time, charging_rate_unit);
+    const auto default_amps_limit =
+        this->device_model->get_optional_value<int>(ControllerComponentVariables::CompositeScheduleDefaultLimitAmps)
```

Comment:
`device_model->get_optional_value<int>` could lead to subtle bugs. It looks like the code will attempt to convert CompositeScheduleDefaultLimitAmps to an int regardless of the actual data type.